kod/object/active/holder/nomoveon/battler/player.kod

```diff
@@ -14171,5 +14178,75 @@ messages:
       return;
    }
 
+  BetaPotionSay(string = $)
+  {
+      local iSchool, iLevel, iAbility, iLearnPoints, iMaxLearnPoints, oLore;
+
+      oLore = Send(SYS,@GetLore);
+      if Send(oLore,@BetaPotionsEnabled) AND Send(Send(SYS,@GetParliament),@BetaPotionsEnabled)  
+         OR Send(SYS, @GetChaosNight)
+      {
+         iLearnPoints = 0;
+         iMaxLearnPoints = Send(Send(SYS, @GetSettings), @GetMaxLearnPoints) + 
+            (Send(self,@GetRawIntellect) * 2) / 5;
+
+         iSchool = send(oLore,@BetaPotionCheckSchool,#string=string);
+         iLevel = send(oLore,@BetaPotionCheckLevel,#string=string);
+         
+         % Default percentage we give for all spells and skills
+         iAbility = 99;
+
+         if iSchool <> 0 AND iLevel <> 0
+         {
+            % 10 is used for weaponcraft
+            if iSchool = SKS_FENCING
+            {
+               send(self,@GivePlayerAllSkills,#school=iSchool,#level=iLevel,#iability=iAbility,#upto=TRUE);
+            }
+            else
+            {
+               if iSchool = SS_SHALILLE
+               {
+                  % Can't add Shal'ille when already has Qor.
+                  if Send(self,@GetNumSpellsInSchool,#school=SS_QOR) 
+                  {
+                     return FALSE;
+                  }
+ 				 
+ 				   piKarma = MAX_GOOD * 100;
```

Comment:
Please change tabs to 3 spaces. In notepad++, go to Settings, Preferences, Tab Settings, then make Tab size 3 and check the Replace by space box. 

Also, please capitalize C calls (i.e. Send, Cons, Post, IsClass etc. but not 'if') and cut line length to 80 characters. Break up statements longer than this at commas or operators (operators go on the following line with the rest of the statement). 2x 'tabs' (6 spaces) for the second line of a Send statement, 1x tab (3 spaces) for an if statement or equation.